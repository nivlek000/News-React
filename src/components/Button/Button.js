import React from 'react';

const Button = ({ children, style, ...props }) => {
    const buttonStyle = {
        border: '1px solid',
        margin: '3px',
        cursor: 'pointer',
        backgroundColor: 'aliceblue',
        color: 'blue',
        padding: '2px 10px',
        ...style


    }


    return (
        <button style={buttonStyle} {...props}>
            {children}
        </button>
    );
}

export default Button;