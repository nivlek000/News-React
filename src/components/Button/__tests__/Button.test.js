import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Button from '../Button';

describe('Button', () => {
    describe('Snapshot', () => {
        it('should create a basic button with defailt styles', () => {
            const components = renderer
                .create(<Button> Click Me </Button>)
                .toJSON();
            expect(components).toMatchSnapshot();

        });

    });

    describe('Unit', () => {
        it('should create a basic button with defailt styles', () => {
            const components = renderer
                .create(<Button style={{ color: "yellow" }}> Click Me </Button>)
                .toJSON();
            expect(components).toMatchSnapshot();

        });

    });
});