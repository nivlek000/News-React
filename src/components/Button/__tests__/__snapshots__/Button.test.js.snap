// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot should create a basic button with defailt styles 1`] = `
<button
  style={
    Object {
      "backgroundColor": "aliceblue",
      "border": "1px solid",
      "color": "blue",
      "cursor": "pointer",
      "margin": "3px",
      "padding": "2px 10px",
    }
  }
>
   Click Me 
</button>
`;

exports[`Button Unit should create a basic button with defailt styles 1`] = `
<button
  style={
    Object {
      "backgroundColor": "aliceblue",
      "border": "1px solid",
      "color": "yellow",
      "cursor": "pointer",
      "margin": "3px",
      "padding": "2px 10px",
    }
  }
>
   Click Me 
</button>
`;
